### PR TITLE
feat(dashboard): regions v2

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- admin V2 regions domain
- list payment providers endpoint
- region payment providers support
- cleanup hooks and remove legacy store hook

### CODE REVIEW

File: packages/admin-next/dashboard/src/v2-routes/regions/region-edit/components/edit-region-form/edit-region-form.tsx [hunk 1]
```diff
@@ -17,48 +15,44 @@ import {
...
 import { formatProvider } from "../../../../../lib/format-provider"
+import { useV2UpdateRegion } from "../../../../../lib/api-v2/region"
 
 type EditRegionFormProps = {
-  region: Region
-  currencies: Currency[]
-  paymentProviders: PaymentProvider[]
-  fulfillmentProviders: FulfillmentProvider[]
+  region: RegionDTO
+  currencies: RegionCountryDTO[]
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/v2-routes/regions/region-edit/components/edit-region-form/edit-region-form.tsx]: todo: Should be currency

**Review by olivermrbl — APPROVED** [review]
LGTM

**Comment by olivermrbl** [comment]
@fPolic, before we merge, let's refactor the client and hooks to follow the pattern introduced in https://github.com/medusajs/medusa/pull/6963

**Comment by fPolic** [comment]
@olivermrbl should be good for review now

**Review by sradevski — APPROVED** [review]

**Pull request merged**